### PR TITLE
HBASE-24373 Implement JvmMetrics in HBase instead of using the one in hadoop

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

**Comment by Apache9** [comment]
Copied the code in hadoop to hbase directly with the same package name, and also added some dummy log4j classes in hbase-logging, it basically worked when running UTs.

But I'm afraid it may not work when starting a cluster, as in tests, maven will make sure that the classes from the project will be loaded before the dependency jars, but when starting a cluster, there is no guarantee which one will be loaded first...

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  1s |  Shelldocs was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 29s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m  0s |  master passed  |
| +1 :green_heart: |  checkstyle  |   2m 20s |  master passed  |
| +1 :green_heart: |  spotbugs  |  25m 33s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   3m 30s |  root in the patch failed.  |
| -0 :warning: |  checkstyle  |   2m 26s |  root: The patch generated 10 new + 203 unchanged - 3 fixed = 213 total (was 206)  |
| +1 :green_heart: |  shellcheck  |   0m  3s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m 52s |  The patch has no ill-formed XML file.  |
| -1 :x: |  hadoopcheck  |   4m 56s |  The patch causes 10 errors with Hadoop v3.1.2.  |
| -1 :x: |  hadoopcheck  |   9m 57s |  The patch causes 10 errors with Hadoop v3.2.1.  |
| -1 :x: |  spotbugs  |   8m 28s |  root in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   5m 55s |  The patch does not generate ASF License warnings.  |
|  |   |  91m 41s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1714 |
| Optional Tests | dupname asflicense shellcheck shelldocs xml hadoopcheck spotbugs hbaseanti checkstyle |
| uname | Linux becd20bd3901 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 5e32e08782 |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-general-check/output/diff-checkstyle-root.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-general-check/output/patch-javac-3.1.2.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-general-check/output/patch-javac-3.2.1.txt |
| spotbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-general-check/output/patch-spotbugs-root.txt |
| Max. process+thread count | 122 (vs. ulimit of 12500) |
| modules | C: hbase-logging hbase-common hbase-metrics-api hbase-metrics hbase-hadoop-compat hbase-client hbase-zookeeper hbase-replication hbase-balancer hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-shell hbase-endpoint hbase-backup hbase-it hbase-rest hbase-examples hbase-shaded hbase-shaded/hbase-shaded-client hbase-shaded/hbase-shaded-client-byo-hadoop hbase-shaded/hbase-shaded-mapreduce hbase-hbtop hbase-assembly hbase-shaded/hbase-shaded-testing-util hbase-shaded/hbase-shaded-testing-util-tester hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants hbase-archetypes/hbase-client-project hbase-archetypes/hbase-shaded-client-project . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) shellcheck=0.4.6 spotbugs=3.1.12 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  7s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 26s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 42s |  master passed  |
| +1 :green_heart: |  compile  |   3m  6s |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m 26s |  branch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 13s |  hbase-logging in master failed.  |
| -0 :warning: |  javadoc  |   0m 13s |  root in master failed.  |
| -0 :warning: |  javadoc  |   0m 14s |  hbase-client-project in master failed.  |
| -0 :warning: |  javadoc  |   0m 14s |  hbase-shaded-client-project in master failed.  |
| -0 :warning: |  javadoc  |   0m 14s |  hbase-asyncfs in master failed.  |
| -0 :warning: |  javadoc  |   0m 19s |  hbase-backup in master failed.  |
| -0 :warning: |  javadoc  |   0m 14s |  hbase-balancer in master failed.  |
| -0 :warning: |  javadoc  |   0m 25s |  hbase-client in master failed.  |
| -0 :warning: |  javadoc  |   0m 14s |  hbase-common in master failed.  |
| -0 :warning: |  javadoc  |   0m 16s |  hbase-examples in master failed.  |
| -0 :warning: |  javadoc  |   0m 17s |  hbase-hadoop-compat in master failed.  |
| -0 :warning: |  javadoc  |   0m 15s |  hbase-hbtop in master failed.  |
| -0 :warning: |  javadoc  |   0m 16s |  hbase-http in master failed.  |
| -0 :warning: |  javadoc  |   0m 19s |  hbase-mapreduce in master failed.  |
| -0 :warning: |  javadoc  |   0m 13s |  hbase-metrics in master failed.  |
| -0 :warning: |  javadoc  |   0m 13s |  hbase-metrics-api in master failed.  |
| -0 :warning: |  javadoc  |   0m 17s |  hbase-procedure in master failed.  |
| -0 :warning: |  javadoc  |   0m 15s |  hbase-replication in master failed.  |
| -0 :warning: |  javadoc  |   0m 19s |  hbase-rest in master failed.  |
| -0 :warning: |  javadoc  |   0m 39s |  hbase-server in master failed.  |
| -0 :warning: |  javadoc  |   0m 56s |  hbase-thrift in master failed.  |
| -0 :warning: |  javadoc  |   0m 15s |  hbase-zookeeper in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   4m  1s |  root in the patch failed.  |
| -1 :x: |  compile  |   2m 35s |  root in the patch failed.  |
| -0 :warning: |  javac  |   2m 35s |  root in the patch failed.  |
| +1 :green_heart: |  shadedjars  |   7m 28s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 13s |  hbase-logging in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 17s |  hbase-common in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 15s |  hbase-metrics-api in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 14s |  hbase-metrics in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 18s |  hbase-hadoop-compat in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 26s |  hbase-client in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 15s |  hbase-zookeeper in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 17s |  hbase-replication in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 15s |  hbase-balancer in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 16s |  hbase-http in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 16s |  hbase-asyncfs in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 16s |  hbase-procedure in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 41s |  hbase-server in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 20s |  hbase-mapreduce in the patch failed.  |
| -0 :warning: |  javadoc  |   1m  0s |  hbase-thrift in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 18s |  hbase-backup in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 19s |  hbase-rest in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 17s |  hbase-examples in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 13s |  hbase-shaded in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 16s |  hbase-hbtop in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 11s |  hbase-shaded-testing-util in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 15s |  hbase-client-project in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 14s |  hbase-shaded-client-project in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 13s |  root in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 198m 44s |  root in the patch failed.  |
|  |   | 254m  3s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1714 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 278f1157444a 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 5e32e08782 |
| Default Java | 2020-01-14 |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-logging.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-root.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-archetypes_hbase-client-project.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-archetypes_hbase-shaded-client-project.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-asyncfs.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-backup.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-balancer.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-client.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-common.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-examples.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-hadoop-compat.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-hbtop.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-http.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-mapreduce.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-metrics.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-metrics-api.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-procedure.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-replication.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-rest.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-server.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-thrift.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-zookeeper.txt |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-mvninstall-root.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-compile-root.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-compile-root.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-logging.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-common.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-metrics-api.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-metrics.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-hadoop-compat.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-client.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-zookeeper.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-replication.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-balancer.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-http.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-asyncfs.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-procedure.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-server.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-mapreduce.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-thrift.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-backup.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-rest.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-examples.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-shaded.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-hbtop.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-shaded_hbase-shaded-testing-util.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-archetypes_hbase-client-project.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-archetypes_hbase-shaded-client-project.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-root.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/testReport/ |
| Max. process+thread count | 2978 (vs. ulimit of 12500) |
| modules | C: hbase-logging hbase-common hbase-metrics-api hbase-metrics hbase-hadoop-compat hbase-client hbase-zookeeper hbase-replication hbase-balancer hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-shell hbase-endpoint hbase-backup hbase-it hbase-rest hbase-examples hbase-shaded hbase-shaded/hbase-shaded-client hbase-shaded/hbase-shaded-client-byo-hadoop hbase-shaded/hbase-shaded-mapreduce hbase-hbtop hbase-assembly hbase-shaded/hbase-shaded-testing-util hbase-shaded/hbase-shaded-testing-util-tester hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants hbase-archetypes/hbase-client-project hbase-archetypes/hbase-shaded-client-project . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 28s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 52s |  master passed  |
| +1 :green_heart: |  compile  |   2m 28s |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m  4s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   9m 46s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   3m 16s |  root in the patch failed.  |
| -1 :x: |  compile  |   2m  2s |  root in the patch failed.  |
| -0 :warning: |  javac  |   2m  2s |  root in the patch failed.  |
| +1 :green_heart: |  shadedjars  |   7m  8s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 13s |  hbase-shaded in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 10s |  hbase-shaded-testing-util in the patch failed.  |
| -0 :warning: |  javadoc  |   1m 46s |  root in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 319m  8s |  root in the patch failed.  |
|  |   | 370m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1714 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 99c2844a7d51 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 5e32e08782 |
| Default Java | 1.8.0_232 |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk8-hadoop3-check/output/patch-mvninstall-root.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk8-hadoop3-check/output/patch-compile-root.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk8-hadoop3-check/output/patch-compile-root.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk8-hadoop3-check/output/patch-javadoc-hbase-shaded.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk8-hadoop3-check/output/patch-javadoc-hbase-shaded_hbase-shaded-testing-util.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk8-hadoop3-check/output/patch-javadoc-root.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/testReport/ |
| Max. process+thread count | 4017 (vs. ulimit of 12500) |
| modules | C: hbase-logging hbase-common hbase-metrics-api hbase-metrics hbase-hadoop-compat hbase-client hbase-zookeeper hbase-replication hbase-balancer hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-shell hbase-endpoint hbase-backup hbase-it hbase-rest hbase-examples hbase-shaded hbase-shaded/hbase-shaded-client hbase-shaded/hbase-shaded-client-byo-hadoop hbase-shaded/hbase-shaded-mapreduce hbase-hbtop hbase-assembly hbase-shaded/hbase-shaded-testing-util hbase-shaded/hbase-shaded-testing-util-tester hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants hbase-archetypes/hbase-client-project hbase-archetypes/hbase-shaded-client-project . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1714/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-logging/src/main/java/org/apache/log4j/Logger.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+package org.apache.log4j;
+
+import org.apache.yetus.audience.InterfaceAudience;
+
+/**
+ * Just for making hadoop work without log4j
+ */
+@InterfaceAudience.Private
+public class Logger {
```
Comment (busbey) [inline hbase-logging/src/main/java/org/apache/log4j/Logger.java:26]: This doesn't end up in our client facing artifacts right?

File: hbase-logging/src/main/java/org/apache/log4j/Logger.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.log4j;
+
+import org.apache.yetus.audience.InterfaceAudience;
+
+/**
+ * Just for making hadoop work without log4j
```
Comment (busbey) [inline hbase-logging/src/main/java/org/apache/log4j/Logger.java:23]: If these are for making Hadoop work, they should be in the hadoop-compat module?

**Comment by Apache9** [comment]
Finally I found it really hard to force JVM to load a specific class when there are conflicts, one possible solution is to add a bootstrap jar but it does not work for java11?

So the solution in #1708 is that, we still have log4j jar on classpath so hadoop is fine, and make sure that there is no slf4j-log4j12 jar on our classpath so all logs will go to log4j2.

PTAL.

Thanks.

**Pull request closed without merging**